web/backend-fastapi/app/rag_states/v4_ImagesAndChunks.py

```diff
@@ -0,0 +1,88 @@
+from typing import List
+from langchain_community.embeddings import HuggingFaceEmbeddings
+from .State import State
+from ..models.neo4j import neo4j_vector_search
+from ..common.chat_objects import ChatHistory
+
+
+class ImagesAndChunks(State):
+    __tag = "v4ImagesAndChunks"  # Don't update this
+    __version = "1"  # Update this if making changes
+    __description = """
+      Uses the same indexing method as UpdatedChunks, but also includes images in the search.
+      """
+
+    __vector_search_query = """
+        CALL db.index.vector.queryNodes($index_name, $top_k, $question) yield node, score
+        OPTIONAL MATCH (node)-[:REFERENCES]->(refNode)
+        RETURN score, 
+                node.ActId AS ActId,  
+                node.RegId as Regulations, 
+                node.sectionId AS sectionId, 
+                node.sectionName AS sectionName, 
+                node.url AS url,
+                node.file_name AS file_name,
+                node.folder AS folder,
+                node.section AS section,
+                node.subfolder AS subfolder,
+                node.type AS type,
+                node.text AS text,
+                collect({
+                    refSectionId: refNode.sectionId, 
+                    refSectionName: refNode.sectionName, 
+                    refActId: refNode.ActId, 
+                    refText: refNode.text
+                }) AS references
+        ORDER BY score DESC
+    """
+    __vector_index = "UpdatedChunksAndImagesv4"
+    __embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
+
+    def __init__(self):
+        super().__init__(
+            self.__tag,
+            self.__version,
+            index=self.__vector_index,
+            query=self.__vector_search_query,
+            kwargs_key="mixtral",
+            description=self.__description,
+        )
+
+    def create_prompt(
+        self, context_str, question, chat_history: List[ChatHistory] = []
+    ) -> str:
+        """
+        Generate a response using the given context and chat history.
+        """
+        chat_history_str = "\n".join(
+            [f"Human: {ch.prompt}\nAI: {ch.response}" for ch in chat_history]
+        )
+        prompt = f"""
```

Comment:
We are duplicating some of these prompts:

do you think having a prompts.json can help?
and then use a format to replace the chat_Histroy or query and referneces inside the json prompts?

Reply:
I experimented with this a bit. 

1. I think having them separate will make future changes easier if you don't want to edit all of them. If we change a centralized prompt to fit a new version, it will have effects on the old versions as well.
2. Using the `.format()` option is really unclear unless your local is filling in those interpolated variables for you. Otherwise, the developer doesn't know for sure what variables it needs without hunting down the prompt.

Opted to leave them for now, but if we see this becoming a maintenance issue, we can come back to it.